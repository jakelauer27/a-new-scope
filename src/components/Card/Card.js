import React, { Component } from 'react'
import propTypes from 'prop-types'
import ResidentsScroller from '../ResidentsScroller/ResidentsScroller'

export default class Card extends Component {
  constructor() {
    super()
    this.state = {
      flipped: false,
      unflipped: false,
      cardObj: null,
    }
  }

  componentDidMount() {
    const { cardData } = this.props
    const cardObj = {
      category: cardData.category,
      favorite: cardData.favorite,
      name: cardData.name,
      type: cardData.type,
      main1Label: '',
      main1: cardData.main1,
      main2Label: '',
      main2: cardData.main2,
      secHeader:'',
      secInfoMain: cardData.secInfoMain,
      secInfoOther: cardData.secInfoOther || '',
    }

    switch(this.props.cardType) {
      case 'people':
        cardObj.main1Label = 'Height'
        cardObj.main1 += ' cm'
        cardObj.main2Label = 'Weight'
        cardObj.main2 += ' kg'
        cardObj.secHeader = 'Homeworld'
        break;
      case 'vehicles':
        cardObj.main1Label = 'Model'
        cardObj.main2Label = 'Class'
        cardObj.secHeader = 'Passengers'
        break;  
      case 'planets':
        cardObj.main1Label = 'Population'
        cardObj.main2Label = 'Climate'
        cardObj.secHeader = 'Residents' 
        break;
      default: break;
    }

    this.setState({ cardObj })
  }


  flipCard = (e) => {
    if(e.target.classList.contains('resident-arrow') 
      || e.target.classList.contains('favorite-btn') ) { return }

    if (this.state.flipped === true) {
      this.setState({
        flipped: false,
        unflipped: true
      })
    } else {
      this.setState({
        flipped: true,
        unflipped: false
      })
    }
  }

  toggleFavorite = (cardObj) => {
    const newCardObj = {...cardObj, favorite: !cardObj.favorite}
    this.setState({cardObj: newCardObj}, () => { 
      this.props.handleStoreData('favorites', newCardObj, newCardObj.favorite, cardObj.category)
    })
  }

  render() {
    const { flipped, unflipped, cardObj } = this.state 
    const { cardType } = this.props
    if (cardObj === null) {
      return (<div></div>)
    } else {
      return (
        <div
          className={`card ${flipped && 'flipped'} ${unflipped && 'unflipped'}`}
          onClick={(e) => this.flipCard(e)}
        >
          <div className='card-front'>
            <div className={`image-side-title card-title card-header-${cardType}`}>
              <h1>{cardObj.name}</h1>
              <i
                className={`fas fa-star ${cardObj.favorite && 'favorite'} favorite-btn`}
                onClick={() => this.toggleFavorite(cardObj)}
              >
              </i>
            </div>
            <img className='front-image' src={`./images/${cardObj.name.replace('\/', '-')}.jpg`} alt='card-image'></img>
          </div>
          <div className={` card-title card-header-${cardType}`}>
            <h1>{cardObj.name}</h1>
            <i
              className={`fas fa-star ${cardObj.favorite && 'favorite'} favorite-btn`}
              onClick={() => this.toggleFavorite(cardObj)}
            >
            </i>
          </div>
          <h1 className='card-type'>{cardObj.type}</h1>
          <div className='stats-container'>
            <div className='stat'>
              <h1>{cardObj.main1Label}</h1>
              <p>{cardObj.main1}</p>
            </div>
            <div className='div-line'></div>
            <div className='stat'>
              <h1>{cardObj.main2Label}</h1>
              <p>{cardObj.main2}</p>
            </div>
          </div>
          <div>
            <div className={`homeworld-title card-header-${cardType}`}>
              <h1 className={`card-header-${cardType}`}>{cardObj.secHeader}</h1>
            </div>
            <ResidentsScroller 
              content1={cardObj.secInfoMain} 
              content2={cardObj.secInfoOther} 
            />
          </div>
          <img className='card-icon' alt='card icon' src={`./images/${cardType}_icon.svg`}></img>
        </div>
      )
    }
  }
}

Card.propTypes = {
  cardType: propTypes.string,
  cardData: propTypes.object
}