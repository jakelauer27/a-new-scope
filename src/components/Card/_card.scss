@mixin title($width, $margin) {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: $margin;
  padding: 5px 15px 5px 15px;
  border-top-right-radius: 25px;
  border-bottom-right-radius: 25px;
  width: $width;
  background: #207D9B;
}

.card {
  background: #353535;
  border-radius: 12px;
  width: 300px;
  height: 270px;
  margin: 0 auto;
  transform: rotate3d(0, 1, 0, 180deg);
  transition: 1.5s;
  position: relative;
  animation: fadeIn 1s ease;

  &::before {
    content: '';
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    background: rgba(0, 0, 0, 0.5);
    opacity: 0;
    z-index: -1;
  }

  h1 {
    font-family: 'Pathway Gothic One';
    letter-spacing: 1px;
    margin: 0;
    color: white;
  }

  i {
    color: #353535;
    font-size: 1.5rem;
    padding-left: 5px;
    transition: .4s ease;
    position: relative;
    z-index: 3;

    &:hover {
      text-shadow: 0 0 8px #FCDD2B;
      transition: .4s ease;
      cursor: pointer;
    }
  }

  .favorite {
    color: #FCDD2B;
  }

  .card-header-people {
    background-color: #207D9B;
  }

  .card-header-vehicles {
    background-color: #4F978A;
  }

  .card-header-planets {
    background-color: #974F72;
  }
}

.card-header-people {
  background-color: #207D9B;
}

.card-header-vehicles {
  background-color: #4F978A;
}

.card-header-planets {
  background-color: #974F72;
}

.card-type {
  text-align: center;
  font-size: 1.2rem;
  font-weight: 200;
}

.div-line {
  background-color: grey;
  height: 46px;
  margin-top: 6px;
  width: 1px;
}

.card-icon {
  position: absolute;
  width: 100px;
  bottom: 10px;
  right: 25px;
}

.card-title {
  @include title(75%, 22px auto 5px 0px);
}

.homeworld-title {
  @include title(35%, 10px auto 5px 0px);
}

.stats-container {
  margin: 10px auto 0px auto;
  display: flex;
  justify-content: space-around;
}

.stat {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  p {
    font-family: 'Pathway Gothic One';
    margin: 5px;
    font-size: 1.2rem;
    color: white;
  }
}

h1.homeworld {
  margin-left: 20px;
}

h1.population {
  margin-left: 30px;
  color: white;
  font-size: 1.2rem;
}

.card-front {
  border-radius: 12px;
  position: absolute;
  z-index: 100;

  .image-side-title {
    @include title(75%, 22px auto 5px 0px);
    position: absolute;
    right: 0px;
    transform: rotate3d(0, 1, 0, 180deg);
  }

  .card-header-people {
    background-color: #207D9B;
  }

  .card-header-vehicles {
    background-color: #4F978A;
  }

  .card-header-planets {
    background-color: #974F72;
  }
}

.front-image {
  border-radius: 12px;
}

.flipped {
  animation: unflipCard 1.5s ease;
  animation-fill-mode: both;

  .card-front {
    animation: fadeOutImage 1.0s ease;
    animation-fill-mode: both;
  }

  .card-title {
    animation: unflipTitle 1.5s ease;
    animation-fill-mode: both;
  }
}

.unflipped {
  animation: flipCard 1.5s ease;
  animation-fill-mode: both;

  .card-front {
    animation: fadeInImage 1.0s ease;
    animation-fill-mode: both;
  }

  .card-title {
    animation: flipTitle 1.5s ease;
    animation-fill-mode: both;
  }
}

@keyframes flipCard {
  0% {
    transform: rotate3d(0, 0, 0, 0);
  }
  100% {
    transform: rotate3d(0, 1, 0, 180deg);
  }
}

@keyframes unflipCard {
  0% {
    transform: rotate3d(0, 1, 0, 180deg);
  }
  100% {
    transform: rotate3d(0, 0, 0, 0);
  }
}

@keyframes fadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

@keyframes fadeOut {
  0% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}

@keyframes fadeInImage {
  0%, 40% {
    opacity: 0;
  }
  41%, 100% {
    opacity: 1;
    z-index: 50
  }
}

@keyframes fadeOutImage {
  0%, 39% {
    opacity: 1;
  }
  40%, 100% {
    opacity: 0;
    z-index: 1;
  }
}

