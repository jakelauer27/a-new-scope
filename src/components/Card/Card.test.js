import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme'

describe('Card', () => {
  let wrapper
  const mockData = {
    name: 'Test Planet',
    type: 'Fun Terrrain',
    main1: '200K',
    main2: 'Warm',
    secInfoMain: ['Jim', 'Bob'],
    secInfoOther: 'None',
    favorite: false,
    category: 'planets'
  }
  let formattedData
  let handleStoreData = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <Card 
        cardType={'planets'} 
        cardData={mockData} 
        handleStoreData={handleStoreData}
    />)
    formattedData = {
      favorite: false,
      category: 'planets',
      name: 'Test Planet',
      type: 'Fun Terrrain',
      main1Label: 'Population',
      main1: '200K',
      main2Label: 'Climate',
      main2: 'Warm',
      secHeader: 'Residents',
      secInfoMain: ['Jim', 'Bob'],
      secInfoOther: 'None'
    }
  })

  it('Should default flipped to false, and unflipped to false', () => {
    wrapper = shallow(<Card cardType={'planets'} cardData={mockData}/>, { disableLifecycleMethods: true})
    const expected = {
      flipped: false,
      unflipped: false,
      cardObj: null
    }

    expect(wrapper.state()).toEqual(expected)
  })

  it('Should render an empty div if cardObj is null', () => {
    wrapper = shallow(<Card cardType={'planets'} cardData={mockData}/>, { disableLifecycleMethods: true})

    expect(wrapper).toMatchSnapshot()
  })

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('ComponentDidMount', () => {

    it('Should set CardObj to reformatted cardData', () => {
      const expected = {
        flipped: false,
        unflipped: false,
        cardObj: formattedData
      }

      expect(wrapper.state()).toEqual(expected)
    })

    it('Should set CardObj labels to vehicle labels if category is vehicle', () => {
      wrapper = shallow(<Card cardType={'vehicles'} cardData={mockData}/>)
      const expected = {
        favorite: false,
        category: 'planets',
        name: 'Test Planet',
        type: 'Fun Terrrain',
        main1Label: 'Model',
        main1: '200K',
        main2Label: 'Class',
        main2: 'Warm',
        secHeader: 'Passengers',
        secInfoMain: ['Jim', 'Bob'],
        secInfoOther: 'None'
      }

      expect(wrapper.instance().state.cardObj).toEqual(expected)
    })

    it('Should set CardObj labels to character labels if category is characters', () => {
      wrapper = shallow(<Card cardType={'people'} cardData={mockData}/>)
      const expected = {
        favorite: false,
        category: 'planets',
        name: 'Test Planet',
        type: 'Fun Terrrain',
        main1Label: 'Height',
        main1: '200K cm',
        main2Label: 'Weight',
        main2: 'Warm kg',
        secHeader: 'Homeworld',
        secInfoMain: ['Jim', 'Bob'],
        secInfoOther: 'None'
      }

      expect(wrapper.instance().state.cardObj).toEqual(expected)
    })

    it('Should not set CardObj labels if there is no Category', () => {
      wrapper = shallow(<Card cardType={''} cardData={mockData}/>)
      const expected = {
        favorite: false,
        category: 'planets',
        name: 'Test Planet',
        type: 'Fun Terrrain',
        main1Label: '',
        main1: '200K',
        main2Label: '',
        main2: 'Warm',
        secHeader: '',
        secInfoMain: ['Jim', 'Bob'],
        secInfoOther: 'None'
      }
      expect(wrapper.instance().state.cardObj).toEqual(expected)
    })

    it.skip('Should fire the toggleFavorite function when the front favorite button is clicked', () => {
      // wrapper.find('.favorite-btn')[0].simulate('click')
    })

    it.skip('Should fire the toggleFavorite function when the back favorite button is clicked', () => {

    })
  })

  describe('flipCard', () => {

    it('should not setState if the click occured on the favorite button', () => {
      const expected = {
        flipped: false,
        unflipped: false,
        cardObj: formattedData
      }

      wrapper.find('.card').simulate('click',  {target: {classList: {contains: () => true}}})

      expect(wrapper.state()).toEqual(expected)
    })

    it('should not setState if the click occured on a scroll button', () => {
      const expected = {
        flipped: false,
        unflipped: false,
        cardObj: formattedData
      }

      wrapper.find('.card').simulate('click',  {target: {classList: {contains: () => true}}})

      expect(wrapper.state()).toEqual(expected)
    })
    
    it('should set flipped to true when unflipped', () => {
      const expected = {
        flipped: true,
        unflipped: false,
        cardObj: formattedData
      }
      
      wrapper.find('.card').simulate('click',  {target: {classList: {contains: () => false}}})

      expect(wrapper.state()).toEqual(expected)
    })
    
    it('should set flipped to false when flipped', () => {
      const expected = {
        flipped: false,
        unflipped: true,
        cardObj: formattedData
      }
      
      wrapper.find('.card').simulate('click', {target: {classList: {contains: () => false}}})
      wrapper.find('.card').simulate('click',  {target: {classList: {contains: () => false}}})

      expect(wrapper.state()).toEqual(expected)
    })
  })

  describe('ToggleFavorite', () => {

    it('Should toggle the favorite property on the card object in state', () => {
      const expectedData = true
      
      wrapper.instance().toggleFavorite(mockData)
      expect(wrapper.state().cardObj.favorite).toEqual(expectedData)
    })

    it('Should store the updated card in local storage', () => {
     
      wrapper.instance().toggleFavorite(mockData)
      expect(handleStoreData).toHaveBeenCalled()
    })
  })
})
