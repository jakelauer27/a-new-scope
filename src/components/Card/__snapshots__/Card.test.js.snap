// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardData={
      Object {
        "main1": "200K",
        "main2": "Warm",
        "name": "Test Planet",
        "secInfoMain": Array [
          "Jim",
          "Bob",
        ],
        "secInfoOther": "None",
        "type": "Fun Terrrain",
      }
    }
    cardType="planets"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-title"
        >
          <h1>
            Test Planet
          </h1>
          <i
            className="fas fa-star"
          />
        </div>,
        <h1
          className="species"
        >
          Fun Terrrain
        </h1>,
        <div
          className="stats-container"
        >
          <div
            className="stat"
          >
            <h1>
              Population
            </h1>
            <p>
              200K
            </p>
          </div>
          <div
            className="stat"
          >
            <h1>
              Climate
            </h1>
            <p>
              Warm
            </p>
          </div>
        </div>,
        <div>
          <div
            className="homeworld-title"
          >
            <h1>
              Residents
            </h1>
          </div>
          <h1
            className="homeworld"
          >
            Jim
            Bob
          </h1>
          <h1
            className="population"
          >
            None
          </h1>
        </div>,
      ],
      "className": "card false false",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Test Planet
            </h1>,
            <i
              className="fas fa-star"
            />,
          ],
          "className": "card-title",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test Planet",
            },
            "ref": null,
            "rendered": "Test Planet",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-star",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Fun Terrrain",
          "className": "species",
        },
        "ref": null,
        "rendered": "Fun Terrrain",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="stat"
            >
              <h1>
                Population
              </h1>
              <p>
                200K
              </p>
            </div>,
            <div
              className="stat"
            >
              <h1>
                Climate
              </h1>
              <p>
                Warm
              </p>
            </div>,
          ],
          "className": "stats-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Population
                </h1>,
                <p>
                  200K
                </p>,
              ],
              "className": "stat",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Population",
                },
                "ref": null,
                "rendered": "Population",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "200K",
                },
                "ref": null,
                "rendered": "200K",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Climate
                </h1>,
                <p>
                  Warm
                </p>,
              ],
              "className": "stat",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Climate",
                },
                "ref": null,
                "rendered": "Climate",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Warm",
                },
                "ref": null,
                "rendered": "Warm",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="homeworld-title"
            >
              <h1>
                Residents
              </h1>
            </div>,
            <h1
              className="homeworld"
            >
              Jim
              Bob
            </h1>,
            <h1
              className="population"
            >
              None
            </h1>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1>
                Residents
              </h1>,
              "className": "homeworld-title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Residents",
              },
              "ref": null,
              "rendered": "Residents",
              "type": "h1",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Jim",
                "Bob",
              ],
              "className": "homeworld",
            },
            "ref": null,
            "rendered": Array [
              "Jim",
              "Bob",
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "None",
              "className": "population",
            },
            "ref": null,
            "rendered": "None",
            "type": "h1",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-title"
          >
            <h1>
              Test Planet
            </h1>
            <i
              className="fas fa-star"
            />
          </div>,
          <h1
            className="species"
          >
            Fun Terrrain
          </h1>,
          <div
            className="stats-container"
          >
            <div
              className="stat"
            >
              <h1>
                Population
              </h1>
              <p>
                200K
              </p>
            </div>
            <div
              className="stat"
            >
              <h1>
                Climate
              </h1>
              <p>
                Warm
              </p>
            </div>
          </div>,
          <div>
            <div
              className="homeworld-title"
            >
              <h1>
                Residents
              </h1>
            </div>
            <h1
              className="homeworld"
            >
              Jim
              Bob
            </h1>
            <h1
              className="population"
            >
              None
            </h1>
          </div>,
        ],
        "className": "card false false",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Test Planet
              </h1>,
              <i
                className="fas fa-star"
              />,
            ],
            "className": "card-title",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Test Planet",
              },
              "ref": null,
              "rendered": "Test Planet",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-star",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Fun Terrrain",
            "className": "species",
          },
          "ref": null,
          "rendered": "Fun Terrrain",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="stat"
              >
                <h1>
                  Population
                </h1>
                <p>
                  200K
                </p>
              </div>,
              <div
                className="stat"
              >
                <h1>
                  Climate
                </h1>
                <p>
                  Warm
                </p>
              </div>,
            ],
            "className": "stats-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Population
                  </h1>,
                  <p>
                    200K
                  </p>,
                ],
                "className": "stat",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Population",
                  },
                  "ref": null,
                  "rendered": "Population",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "200K",
                  },
                  "ref": null,
                  "rendered": "200K",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Climate
                  </h1>,
                  <p>
                    Warm
                  </p>,
                ],
                "className": "stat",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Climate",
                  },
                  "ref": null,
                  "rendered": "Climate",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Warm",
                  },
                  "ref": null,
                  "rendered": "Warm",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="homeworld-title"
              >
                <h1>
                  Residents
                </h1>
              </div>,
              <h1
                className="homeworld"
              >
                Jim
                Bob
              </h1>,
              <h1
                className="population"
              >
                None
              </h1>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1>
                  Residents
                </h1>,
                "className": "homeworld-title",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Residents",
                },
                "ref": null,
                "rendered": "Residents",
                "type": "h1",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Jim",
                  "Bob",
                ],
                "className": "homeworld",
              },
              "ref": null,
              "rendered": Array [
                "Jim",
                "Bob",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "None",
                "className": "population",
              },
              "ref": null,
              "rendered": "None",
              "type": "h1",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Card Should render an empty div if cardObj is null 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardData={
      Object {
        "main1": "200K",
        "main2": "Warm",
        "name": "Test Planet",
        "secInfoMain": Array [
          "Jim",
          "Bob",
        ],
        "secInfoOther": "None",
        "type": "Fun Terrrain",
      }
    }
    cardType="planets"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
